### ISSUE TITLE
Can't convert `Categorical` to 'datetime64[ns, MET]', but can do it with Series

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#28448]
#### Code Sample, a copy-pastable example if possible

```python
from pandas import Categorical

Categorical(['2015-01-01']).astype('datetime64[ns]')  # works
Categorical(['2015-01-01']).astype('datetime64[ns, MET]')  # fails
```

This is because the `astype` method returns
```
       np.array(self, dtype=dtype, copy=copy)
```
, and `numpy` arrays don't handle this dtype.

Is this expected behaviour?

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-60-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 0.25.1
numpy            : 1.16.4
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : None
pytest           : 5.0.1
hypothesis       : None
sphinx           : 2.2.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.8.0
pandas_datareader: None
bs4              : 4.8.0
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : 2.7.0
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.13.0
pytables         : None
s3fs             : None
scipy            : 1.2.1
sqlalchemy       : None
tables           : 3.5.2
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None

</details>

**Comment by MarcoGorelli** [comment]
Seems to me like this happens because
```
>>> from pandas.core.dtypes.common import is_extension_type
>>> is_extension_type('datetime64[ns]')
False
>>> is_extension_type('datetime64[ns, MET]')
```
and so then, in `blocks.py`, in the latter (tz-aware) case we get to
```
            if self.is_extension:
                values = self.values.astype(dtype)
            else:
```
, which fails, while in the first case this `if` statement returns False.

Is this expected behaviour?

**Comment by WillAyd** [comment]
I think this should be consistent so probably OK to support. If you have a PR you'd like to submit would certainly welcome one

**Comment by MarcoGorelli** [comment]
Fixed by #28668

**Comment by jreback** [comment]
hmm do we have a test for this?
